FabLabKasse/faucardPayment/generateLog.py

```diff
@@ -0,0 +1,131 @@
+    parser.add_argument('-f', "--file", help="MagPosLog to build log from", required=True)
+    parser.add_argument('-s', "--startdate", help="The Start Date - format YYYY-MM-DD_HH:MM:SS", required=True, type=valid_date)
+    parser.add_argument('-e', "--enddate", help = "The End Date - format YYYY-MM-DD_HH:MM:SS", required=False, type=valid_date)
+    parser.add_argument('-o', "--outputpath", help="Output path of csv and summary, e.g. /usr/var/test -> test.csv, testsummary.txt", required=True)
+    parser.add_argument('-k', "--kassenbuch", help="Kassenbuch to build log from", required=True)
+
+    try:
+        args = parser.parse_args()
+    except argparse.ArgumentTypeError as e:
+        print "ERROR: ArgumentTypeError '{0}'".format(e)
+        print "Please refer argument format to given example in -help"
+        raise
+    except argparse.ArgumentError as e:
+        print "ERROR: ArgumentParsing failed '{0}'".format(e)
+        print "Please check if all arguments are valid"
+        raise
+
+    startdate = args.startdate
+    enddate = args.enddate
+    summe = Decimal(0)
+    cfg = scriptHelper.getConfig()
+
+    try:
+        con = sqlite3.connect(args.file)
+        cur = con.cursor()
+        con.text_factory = unicode
+        cur.execute("SELECT timestamp_payed, cardnumber, oldbalance, amount, newbalance,  datum FROM MagPosLog WHERE timestamp_payed >= ? AND timestamp_payed <= ? ORDER BY timestamp_payed ASC",(startdate, enddate))
+
+
+        conKb = sqlite3.connect(args.kassenbuch)
+        curKb = conKb.cursor()
+        conKb.text_factory = unicode
+    except sqlite3.OperationalError as e:
+        print "ERROR: {0}".format(e)
+        raise
+
+
+    try:
+        # Open csv filed
+        outputfile = codecs.open(args.outputpath+".csv", 'w', encoding="utf-8")
+
+        # Write header
+        outputfile.write(u"Zeitstempel Zahlung, Kartennummer, Old Balance, Zahlungsbetrag, New Balance, Rechnung\n")
+        outputfile.write(u",,,,,\n")
+
+        firstdate = None
+        lastdate = None
+        rechnungsliste = []
+
+        counter = 0
+        # Write one row for each row in database
+        for row in cur.fetchall():
+            timestamp = datetime.strptime(row[0].split(".")[0],"%Y-%m-%d %H:%M:%S")
+
+            if counter == 0:
+                firstdate = timestamp
+                counter += 1
+            else:
+                lastdate = timestamp
+
+            # Determine corresponding rechnung
+            curKb.execute("SELECT rechnung FROM buchung WHERE (betrag = ? OR betrag = ?) AND datum BETWEEN ? AND ? ",(unicode(row[3]), "{:.3f}".format(row[3]), timestamp, timestamp + timedelta(seconds=20)))
+            safetyCounter = 0
+            rechnungsnr = -1
+
+            for rowKb in curKb.fetchall():
+                safetyCounter += 1
+                rechnungsnr = rowKb[0]
+
+            #assert safetyCounter == 1
```

Comment:
There should only be one matching entry in buchung. The safetyCounter counts the entries which meet the requirements in the sql statement. I think i might commented it out while testing the script. 
This should be uncommented.